components/error-boundary.tsx

```
@@ -0,0 +1,51 @@
+    if (this.state.hasError) {
+      // You can render any custom fallback UI.
+      return (
+        <div className="flex flex-col items-center justify-center h-screen bg-background text-foreground p-4 text-center">
+          <h1 className="text-2xl font-bold mb-4">Something went wrong.</h1>
+          <p className="text-muted-foreground">
+            An unexpected error occurred. We&apos;ve logged the issue and will look into it.
+          </p>
+          <button
+            onClick={() => this.setState({ hasError: false })}
+            className="mt-6 px-4 py-2 bg-primary text-primary-foreground rounded-md hover:bg-primary/90"
+          >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)